File: sdk/python/src/chainlogistics_sdk/services/stats.py
"""Stats service for accessing statistics and analytics."""

from __future__ import annotations

from typing import TYPE_CHECKING

from ..models import (
    DbHealthResponse,
    GlobalStats,
    HealthResponse,
    LivenessResponse,
    MonitoringHealthResponse,
    ReadinessResponse,
)

if TYPE_CHECKING:
    from ..client import ChainLogisticsClient


class StatsService:
    """Service for accessing statistics and analytics."""
    
    def __init__(self, client: ChainLogisticsClient):
        """Initialize the service.
        
        Args:
            client: ChainLogistics client instance
        """
        self.client = client
    
    def get_global(self) -> GlobalStats:
        """Get global statistics.
        
        Returns:
            Global statistics
        """
        data = self.client.get("api/v1/stats")
        return GlobalStats(**data)
    
    def health(self) -> HealthResponse:
        """Get system health status.
        
        Returns:
            Health response
        """
        data = self.client.get("health")
        return HealthResponse(**data)

    def readiness(self) -> ReadinessResponse:
        """Get readiness status."""
        data = self.client.get("health/ready")
        return ReadinessResponse(**data)

    def liveness(self) -> LivenessResponse:
        """Get liveness status."""
        data = self.client.get("health/live")
        return LivenessResponse(**data)
    
    def db_health(self) -> DbHealthResponse:
        """Get database health status.
        
        Returns:
            Database health response
        """
        data = self.client.get("health/db")
        return DbHealthResponse(**data)

    def monitoring_health(self) -> MonitoringHealthResponse:
        """Get monitoring health status."""
        data = self.client.get("health/monitoring")
        return MonitoringHealthResponse(**data)

File: sdk/rust/src/stats.rs
use crate::{
    client::HttpClient,
    models::GlobalStats,
    Config, Result,
};

/// Service for accessing statistics and analytics
#[derive(Debug, Clone)]
pub struct StatsService {
    client: HttpClient,
}

impl StatsService {
    pub(crate) fn new(client: reqwest::Client, config: Config) -> Self {
        Self {
            client: HttpClient::new(client, config),
        }
    }

    /// Get global statistics
    pub async fn get_global(&self) -> Result<GlobalStats> {
        let request = self.client.get("api/v1/stats");
        self.client.execute(request).await
    }

    /// Get system health status
    pub async fn health(&self) -> Result<crate::models::HealthResponse> {
        let request = self.client.get("health");
        self.client.execute(request).await
    }

    /// Get application readiness status
    pub async fn readiness(&self) -> Result<crate::models::ReadinessResponse> {
        let request = self.client.get("health/ready");
        self.client.execute(request).await
    }

    /// Get process liveness status
    pub async fn liveness(&self) -> Result<crate::models::LivenessResponse> {
        let request = self.client.get("health/live");
        self.client.execute(request).await
    }

    /// Get database health status
    pub async fn db_health(&self) -> Result<crate::models::DbHealthResponse> {
        let request = self.client.get("health/db");
        self.client.execute(request).await
    }

    /// Get monitoring health status
    pub async fn monitoring_health(&self) -> Result<crate::models::MonitoringHealthResponse> {
        let request = self.client.get("health/monitoring");
        self.client.execute(request).await
    }
}
